app/src/main/java/com/example/campusguide/directions/RoutePreviewActivity.kt

```diff
@@ -0,0 +1,146 @@
+
+        stepInstruction.text = parseHTMLString(steps[0].htmlInstruction)
+        previousStepButton.isEnabled = false
+        stepInstruction.setCompoundDrawablesWithIntrinsicBounds(R.drawable.ic_turn_right, 0, 0, 0)
+
+        initializer.setOnMapReadyListener {
+            val path = PathPolyline(pathPolyline.getStart(), pathPolyline.getEnd(), pathPolyline.getPath())
+            setPathOnMapAsync(path)
+            setCurrentStep(0)
+            focusCameraOnCurrentStep(PathPolyline("", "", currentStepPath), 0)
+        }
+
+        previousStepButton.setOnClickListener {
+            previousStep(stepInstruction)
+        }
+
+        nextStepButton.setOnClickListener {
+            nextStep(stepInstruction)
+        }
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        finish()
+        return true
+    }
+
+    private fun previousStep(textView: TextView) {
+        i--
+        textView.text = parseHTMLString(steps[i].htmlInstruction)
+        setIcon(textView, i)
+        setCurrentStep(i)
+        focusCameraOnCurrentStep(
+            PathPolyline("", "", currentStepPath),
+            i
+        )
+        if (i == 0)
+            previousStepButton.isEnabled = false
+        else
+            nextStepButton.isEnabled = true
+    }
+
+    private fun nextStep(textView: TextView) {
+        i++
+        textView.text = parseHTMLString(steps[i].htmlInstruction)
+        setIcon(textView, i)
+        setCurrentStep(i)
+        focusCameraOnCurrentStep(
+            PathPolyline("", "", currentStepPath),
+            i
+        )
+        if (i == steps.size - 1) {
+            nextStepButton.isEnabled = false
+        } else {
+            previousStepButton.isEnabled = true
+        }
+    }
+
+    private fun setPathOnMapAsync(path: PathPolyline) {
+        GlobalScope.launch {
+            path.waitUntilCreated()
+            runOnUiThread {
+                map.addPath(path)
+            }
+        }
+    }
+
+    private fun focusCameraOnCurrentStep(path: PathPolyline, step: Int) {
+        GlobalScope.launch {
+            path.waitUntilCreated()
+            runOnUiThread {
+                map.moveCamera(path.getPathBounds())
+            }
+        }
+    }
+
+    private fun parseHTMLString(string: String): String {
+        return Jsoup.parse(string).text()
+    }
+
+    private fun setCurrentStep(i: Int) {
+        currentStepPath.clear()
+        var startLatLng = LatLng(steps[i].startLocation.lat.toDouble(),
+            steps[i].startLocation.lng.toDouble())
+        var endLatLng = LatLng(steps[i].endLocation.lat.toDouble(),
+            steps[i].endLocation.lng.toDouble())
+        currentStepPath.add(startLatLng)
+        currentStepPath.add(endLatLng)
+    }
+
+    private fun setIcon(instruction: TextView, step: Int) {
+        val drawable: Int =
+            when (steps[step].maneuver) {
+                "turn-right" -> R.drawable.ic_turn_right
```

Comment:
These string values could be put into constants

Reply:
I'll do that, thank you